### ISSUE TITLE
Render To Texture Example Panics on WebGL

### ISSUE THREAD

**Issue by LennyPhoenix** [gfx-rs/wgpu#4266]
**Description**
Attempting to run the render-to-texture example using WebGPU seems to panic on the client-side:

[localhost-1697667788390.log](https://github.com/gfx-rs/wgpu/files/13034551/localhost-1697667788390.log)

It appears that the returned adapter is `None`.

**Repro steps**
Clone the repo on the `trunk` branch, then run `cargo xtask run-wasm --bin render-to-texture`.

**Expected vs observed behavior**
Adapter to not be `None`, and example to run.

**Extra materials**
![image](https://github.com/gfx-rs/wgpu/assets/57148878/424d7f57-5703-4230-bdc0-713ff9209c1e)

![image](https://github.com/gfx-rs/wgpu/assets/57148878/0f1f3c24-6b08-40d1-b625-73edeb852a0b)

**Platform**
Arch Linux, Rust v1.70.0

**Comment by cwfitzgerald** [comment]
This appears to be an issue with your WebGPU implementation. Note the warning "failed to create WebGPU context provider". WebGPU isn't supported on linux on chrome yet, so if you want to run the examples, you need to run them with `--features webgl`

**Comment by LennyPhoenix** [comment]
I'm getting the same error even with `--features webgl`: (`cargo xtask run-wasm --bin render-to-texture --features webgl`)
![image](https://github.com/gfx-rs/wgpu/assets/57148878/d0773969-3a90-4188-a0de-d75caa3d4bd3)

**Comment by cwfitzgerald** [comment]
Ah fun, alright :)

**Comment by cwfitzgerald** [comment]
Could you check to make sure https://webglreport.com/?v=2 reports no problem with webgl2?

**Comment by LennyPhoenix** [comment]
Seems happy:
![image](https://github.com/gfx-rs/wgpu/assets/57148878/01d6e75a-197d-4796-a04c-099f9091bd9c)

**Comment by cwfitzgerald** [comment]
Oh, the reason is that render-to-texture only works with WebGPU right now.

**Comment by LennyPhoenix** [comment]
I see, so is there no way to get an adapter with `compatible_surface` set to `None` on linux?

**Comment by Merlin-Brandt** [comment]
Same issue here on Linux Mint 21.2 and Chrome 117.0.5938.132.

**Comment by cwfitzgerald** [comment]
On WebGL2, we need to get the WebGL2RenderingContext _from_ a canvas, so you need to register a canvas surface with us before you enumerate adapters so we have a WebGL2RenderingContext to pull from.

This example does not do this, so only works on WebGPU.

**Comment by LennyPhoenix** [comment]
Makes sense, I can confirm that after enabling experimental WebGPU support (`brave --enable-unsafe-webgpu --enable-features=Vulkan,UseSkiaRenderer`) the example works fine.

**Comment by zetanumbers** [comment]
I have modified render_to_texture example and it works fine with WebGL. I had to pretty much just lower required limits and use a canvas for the adapter request.
